File: 1776-minimum-operations-to-reduce-x-to-zero/1776-minimum-operations-to-reduce-x-to-zero.js
/**
 * @param {number[]} nums
 * @param {number} x
 * @return {number}
 */
var minOperations = function(nums, x) {
    let sum = nums.reduce((a,b)=>a+b,0);
    sum -= x;
    if(sum === 0) return nums.length;
    let result = 0;
    let left = 0;
    let now = 0;
    for(let right=0; right<nums.length; right++){
        now += nums[right];
        while(left <= right && now > sum){
            now -= nums[left];
            left++;
        }
        if(now === sum){
            result = Math.max(result, right-left+1);
        }
    }
    return result ? nums.length - result : -1;
};

File: 1776-minimum-operations-to-reduce-x-to-zero/1776-minimum-operations-to-reduce-x-to-zero.java
class Solution {
    public int minOperations(int[] nums, int x) {
        int sum = 0;
        for(int num : nums){
            sum += num;
        }
        sum-=x;
        if(sum ==0) return nums.length;
        int left = 0;
        int now = 0;
        int result = 0;
        for(int right=0; right<nums.length; right++){
            now += nums[right];
            while(left <= right && now > sum){
                now -= nums[left];
                left++;
            }
            if(sum == now){
                result = Math.max(result, right - left +1);
            }
        }
        return result !=0? nums.length - result:-1;
    }
}